Source/WebKit/Scripts/PreferencesTemplates/WebFeatures.h.erb

```diff
@@ -0,0 +1,65 @@
+/*
+ * <%= @warning %>
+ *
+ * Copyright (C) 2024 Apple Inc. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. AND ITS CONTRIBUTORS ``AS IS''
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+ * THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL APPLE INC. OR ITS CONTRIBUTORS
+ * BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+ * THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#pragma once
+
+#include "ArgumentCoders.h"
+
+namespace WebKit {
+
+class WebPreferences;
+
+struct WebFeatures {
+<% for @pref in @exposedFeatures do -%>
+<%- if @pref.condition -%>
+#if <%= @pref.condition %>
+<%- end -%>
+    bool <%= @pref.name %> : 1 { false };
+<%- if @pref.condition -%>
+#endif
+<%- end -%>
+<%- end -%>
+
+    WebFeatures() = default;
+    bool operator==(const WebFeatures&) const = default;
+
+    explicit WebFeatures(const WebPreferences&);
+};
+
+} // namespace WebKit
+
+namespace IPC {
+
+class Encoder;
+class Decoder;
+
+template<> struct ArgumentCoder<WebKit::WebFeatures> {
```

Comment:
Does TestWebKitAPI.IPCTestingAPI.SerializedTypeInfo pass after this in debug builds?  I think we probably need to generate a serialization.in file instead.

Reply:
Changed it to generate serialization.in file.